server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -236,18 +254,135 @@ public void build(
             String dimension = dimensionsSplitOrder.get(i).getField();
             FieldInfo dimensionFieldInfo = writeState.fieldInfos.fieldInfo(dimension);
             if (dimensionFieldInfo == null) {
-                dimensionFieldInfo = getFieldInfo(dimension, DocValuesType.SORTED_NUMERIC);
+                // TODO : make this better
```

Comment:
nit:: what are we looking for here exactly?

Reply:
I think i did add docValuesType to dimension , so I can remove this